hudi-common/src/main/java/org/apache/hudi/common/util/CleanerUtils.java

```diff
@@ -168,10 +168,18 @@ public static Option<HoodieInstant> getEarliestCommitToRetain(
   public static HoodieCleanerPlan getCleanerPlan(HoodieTableMetaClient metaClient, HoodieInstant cleanInstant)
       throws IOException {
     CleanPlanMigrator cleanPlanMigrator = new CleanPlanMigrator(metaClient);
+    cleanInstant = getCleanRequestInstant(metaClient, cleanInstant);
```

Comment:
can we make the `cleanInstant` as `REQUESTED` from the invoker?

Reply:
> can we make the `cleanInstant` as `REQUESTED` from the invoker?

The current invoker does not care whether the incoming Instant is request or inflight. To be compatible with the current usage logic of invoker, I choose to perform the instant conversion at this method. Moreover, from the design of this method itself, given a clean instant of any state, It is reasonable that we should all be able to return to the correct plan.

Reply:
so there is possibility an empty inflight instant been removed for 2 times.

Reply:
> so there is possibility an empty inflight instant been removed for 2 times.

If there is a concurrent cleaner, it is indeed possible, but this behavior has nothing to do with the current code modification

Reply:
Can we check all the callers of it, I see several places have already do some pre-transformation to make sure the instant is a `REQUESTED` before calling it.